### PR TITLE
Get parcel shop list

### PR DESCRIPTION
**Pull request by lukasz-falda**
Very similar to https://github.com/answear/overseas-bundle.

AC
---
* can get parcel shop list

### CODE REVIEW

File: src/Service/ParcelShopsService.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        ?bool $codHandler = null,
+        ?bool $dropOff = null
+    ): array {
+        $rawParcelShopCollection = $this->getList(
+            $withPclShopIn,
+            $withParcelLockIn,
+            $senderId,
+            $codHandler,
+            $dropOff
+        );
```
Comment (lukasz-falda) [inline src/Service/ParcelShopsService.php:48]: get parcels by `getList` method

File: src/Service/ParcelShopsService.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+        $parcelShopCollection = [];
+        foreach ($rawParcelShopCollection as $rawParcelShop) {
+            $openings = $this->getOpenings($rawParcelShop->pclshopid);
+
+            $parcelShop = ParcelShop::fromRawParcelShop($rawParcelShop, $openings);
```
Comment (lukasz-falda) [inline src/Service/ParcelShopsService.php:54]: for all parcels get openings and prepare final ParcelShop object

File: src/Enum/ParcelShopAction.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    public const GET_LIST = 'getList';
+    public const GET_LNG2 = 'getLng2';
+    public const GET_OPENINGS = 'getOpenings';
+
+    public static function getOpenings(): self
+    {
+        return static::get(static::GET_OPENINGS);
+    }
+
+    public static function getLng2(): self
```
Comment (ropczan) [inline src/Enum/ParcelShopAction.php:20]: What does `getLng2` mean, are we able to give some clearer name?

File: src/Request/GetParcelShops.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        $urlQuery = http_build_query(
+            [
+                'action' => $this->action->getValue(),
+                'ctrcode' => null === $this->countryCode ? null : $this->countryCode->getValue(),
+                'senderid' => $this->senderId ?? null,
+                'pclshopin' => null === $this->pclShopIn ? null : (int) $this->pclShopIn,
+                'parcellockin' => null === $this->parcelLockIn ? null : (int) $this->parcelLockIn,
+                'codhandler' => null === $this->codHandler ? null : (int) $this->codHandler,
+                'dropoff' => null === $this->dropOff ? null : (int) $this->dropOff,
+                'pclshopid' => null === $this->shopId ? null : $this->shopId,
```
Comment (ropczan) [inline src/Request/GetParcelShops.php]: Null Coalescing Operator :thinking:

File: src/Serializer/Serializer.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    public function decodeResponse(string $class, ResponseInterface $response)
+    {
+        return $this->getSerializer()->deserialize(
+            $response->getBody()->getContents(),
+            $class,
+            self::FORMAT
+        );
+    }
+
+    private function getSerializer(): SymfonySerializer
```
Comment (ropczan) [inline src/Serializer/Serializer.php:39]: Why a private method instead of a constructor?

File: src/Enum/ParcelShopAction.php (hunk 3 above)
Comment (lukasz-falda) [reply src/Enum/ParcelShopAction.php:20]: `getLng2` is the action name - this is specific action for `/psmap/psmap_getdata.php`

File: src/Serializer/Serializer.php (hunk 5 above)
Comment (lukasz-falda) [reply src/Serializer/Serializer.php:39]: creating a Symfony Serializer on the constructor creates it on every request - unnecessary and redundant

**Review by ropczan — APPROVED** [review]

**Review by konradkozaczenko — APPROVED** [review]

**Pull request merged**